### PR TITLE
Idle during long arcs

### PR DESCRIPTION
**Pull request by AnHardt**
Idle during long arcs
to prevent from watchdog resets during high segmented fast arcs.

Includes:   Make arc support (G2/G3) configurable #3747

Fix for:   Crash (Hard Reset) when issuing G2/G3 command #3729

### CODE REVIEW

**Comment by jbrazio** [comment]
Will close #3747, if needed we can cherry-pick from this PR.

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -6484,7 +6488,7 @@ void process_next_command() {
         break;
 
       // G2, G3
-      #if DISABLED(SCARA)
+      #if ENABLED(ARC_SUPPORT) & DISABLED(SCARA)
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Oops. Two ampersands would be better here.

**Comment by thinkyhead** [comment]
It's hard to perceive the changes, because indentation and changes are combined in a single commit. Would it be possible to add an intermediate commit, with the code changes but not the indentation, and then a commit with the indentation afterward? If that sounds tedious and annoying, I don't mind doing it myself. It just makes it much easier to see the smaller changes that were made, especially for future reference…. esp. as I will need to add similar changes to the Bézier curve support.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -7514,8 +7514,16 @@ void prepare_move() {
 
     float feed_rate = feedrate * feedrate_multiplier / 60 / 100.0;
 
+    millis_t previous_ms = millis();
```
Comment (jbrazio) [inline Marlin/Marlin_main.cpp]: You can shave this `millis()` call and init the var as 0.

Comment (jbrazio) [inline Marlin/Marlin_main.cpp]: You can shave this `millis()` call and init the var as 0.

**Comment by AnHardt** [comment]
https://github.com/MarlinFirmware/Marlin/pull/3747/files?w=1 
https://github.com/MarlinFirmware/Marlin/pull/3751/commits/89d990800b51d72c56ab92d63b63c3c69a9d2055
Or if you want bot commits togeter but without the changes in the white space
https://github.com/MarlinFirmware/Marlin/pull/3751/files?w=1
?w=1 is your friend. :-)

File: Marlin/Marlin_main.cpp [hunk 3]
```diff
@@ -7423,147 +7427,157 @@ void prepare_move() {
...
-      // Compute exact location by applying transformation matrix from initial radius vector(=-offset).
-      // To reduce stuttering, the sin and cos could be computed at different times.
-      // For now, compute both at the same time.
-      cos_Ti = cos(i * theta_per_segment);
-      sin_Ti = sin(i * theta_per_segment);
-      r_X = -offset[X_AXIS] * cos_Ti + offset[Y_AXIS] * sin_Ti;
-      r_Y = -offset[X_AXIS] * sin_Ti - offset[Y_AXIS] * cos_Ti;
-      count = 0;
-    }
+    millis_t previous_ms = millis();
```
Comment (jbrazio) [inline Marlin/Marlin_main.cpp:7517]: You can shave this `millis()` call and init the var as 0.

Comment (AnHardt) [inline Marlin/Marlin_main.cpp:7517]: Why should i want that.
That would call `idle()` one time at the beginning of the loop, where we don't need it.

File: Marlin/Marlin_main.cpp (hunk 1 above)
Comment (AnHardt) [inline Marlin/Marlin_main.cpp]: Changed.

File: Marlin/Marlin_main.cpp (hunk 3 above)
Comment (jbrazio) [inline Marlin/Marlin_main.cpp:7517]: It depends on the perspective..
It was a suggestion.

**Comment by thinkyhead** [comment]
I'm curious. Why do we disallow arc support with `SCARA`? And it looks like actually we should add a sanity-check for `ARC_SUPPORT` plus `SCARA` now, if they are indeed incompatible.

**Pull request merged**